File: factory-method/c#/Meats/GrilledChicken.cs
using System;

namespace FactoryMethod
{
    public class GrilledChicken : ICookedMeat
    {
        public decimal Weight { get; set; }
        public string Flavouring { get; set; }

        public void Eat()
        {
            Console.WriteLine("Ate the grilled chicken - yum.");
        }

    }
}

File: factory-method/typescript/src/Meats/GrilledChicken.ts
import {ICookedMeat} from "Interfaces/ICookedMeat";
export class GrilledChicken implements ICookedMeat{
  //The compiler doesn't add the property to the 
  // object if it doesn't get set to a value
  // so calling GrilledChicken.Flavouring would
  // return undefined.
  public Weight : number = 0;
  public Flavouring : string = ""; 
  public Eat(): void{
    console.log("Ate the grilled chicken - yum.");
  }
}